### PR TITLE
Turborepo

### PR DESCRIPTION
**Pull request by cdedreuille**

### CODE REVIEW

File: apps/frontpage/app/layout.tsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
+import './globals.css';
 import type { Metadata } from 'next';
-import { Nunito_Sans as FontSans } from 'next/font/google';
-import { cn } from '@utils';
+import './globals.css';
```
Comment (kylegach) [inline apps/frontpage/app/layout.tsx:3]: Duplicate import?

File: apps/frontpage/components/docs/mdx/code-snippets/utils/get-metadata.ts [hunk 2]
```diff
@@ -56,6 +56,7 @@ export const getMetadata = async ({ path, activeVersion }: Props) => {
           const [key, value] = match
             .split('=')
             .map((part) => part.replace(/"/g, ''));
+          // @ts-ignore
```
Comment (kylegach) [inline apps/frontpage/components/docs/mdx/code-snippets/utils/get-metadata.ts:59]: Nit: Better to use `@ts-expect-error` (which will throw an error if it is no longer needed for some reason) and provide an explanation.

File: apps/frontpage/components/home/illustrated-feature-list.tsx [hunk 3]
```diff
@@ -70,14 +70,14 @@ export const IllustratedFeatureList: FC<IllustratedFeatureListProps> = ({
...
-            : 'order-2 md:-mr-12 md:rounded-tr-none md:-border-br-none'
+            : 'order-2 md:-mr-12 md:rounded-tr-none md:-border-br-none',
         )}
         style={{ backgroundColor: bgColor }}
       >
         <AnimatePresence initial={false} custom={direction}>
           <motion.div
             className="absolute top-0 bottom-0 left-0 right-0"
-            key={activeFeature.title}
+            key={activeFeature?.title}
```
Comment (kylegach) [inline apps/frontpage/components/home/illustrated-feature-list.tsx:80]: Better, I think, to cast the type of `activeFeature`:

```ts
const activeFeature = features[activeIndex] as FeatureItem;
```

That assumes that `activeIndex` will always be an index in the `features` array, but that seems safe to me.

Then you can rely on `activeFeature`'s type throughout this code.

File: apps/frontpage/components/home/share/integrations-carousel.tsx [hunk 4]
```diff
@@ -24,14 +24,14 @@ export const IntegrationsCarousel: FC<IntegrationsCarouselProps> = ({
 
   return (
     <div ref={ref} className={className}>
-      <figure className="m-0">{activeIntegration.media}</figure>
+      <figure className="m-0">{activeIntegration?.media}</figure>
```
Comment (kylegach) [inline apps/frontpage/components/home/share/integrations-carousel.tsx:27]: Same as https://github.com/storybookjs/web/pull/34/files#r1592819896

File: apps/frontpage/docs-versions.ts [hunk 5]
```diff
@@ -1,21 +1,29 @@
+export type DocsVersion = {
```
Comment (kylegach) [inline apps/frontpage/docs-versions.ts:1]: For @kylegach — Confirm this still works after CJS → ESM and location changes.

Comment (kylegach) [reply apps/frontpage/docs-versions.ts:1]: It does not: https://github.com/storybookjs/web/pull/34/files#diff-f81db2672a54c32bb8dea3ef9d967f490626798d37ea4b3c59e1b10ad09dbc15R1

File: apps/frontpage/postcss.config.mjs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('postcss-load-config').Config} */
```
Comment (kylegach) [inline apps/frontpage/postcss.config.mjs:1]: `.mjs`?

File: apps/frontpage/scripts/getLocalDocs.ts [hunk 7]
```diff
@@ -50,15 +50,15 @@ async function fetchAndExtract(version: DocsVersion) {
               {
                 strip: 2,
                 C: path.join(__dirname, `../content/docs/${version.id}`),
-                filter: (path) =>
+                filter: (path: any) =>
```
Comment (kylegach) [inline apps/frontpage/scripts/getLocalDocs.ts:53]:
```suggestion
                filter: (path: string) =>
```

File: package.json [hunk 8]
```diff
@@ -1,112 +1,24 @@
 {
-  "name": "storybook-web",
-  "version": "0.0.0",
-  "license": "MIT",
-  "scripts": {
-    "dev": "npx nx run frontpage:dev",
-    "build": "npx nx run frontpage:build"
-  },
+  "name": "my-turborepo",
```
Comment (kylegach) [inline package.json:2]:
```suggestion
  "name": "storybook-web",
```

File: package.json [hunk 9]
```diff
@@ -1,112 +1,24 @@
...
-  }
+    "@repo/eslint-config": "*",
+    "@repo/typescript-config": "*",
+    "prettier": "^3.2.5",
+    "turbo": "^1.13.3"
+  },
+  "engines": {
+    "node": ">=18"
+  },
+  "packageManager": "npm@10.7.0",
```
Comment (kylegach) [inline package.json:19]: Was moving from `pnpm` to `npm` necessary?

File: packages/eslint-config/library.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    // Ignore dotfiles
+    ".*.js",
+    "node_modules/",
+    "dist/",
+  ],
+  overrides: [
+    {
+      files: ["*.js?(x)", "*.ts?(x)"],
+    },
+  ],
```
Comment (kylegach) [inline packages/eslint-config/library.js:33]: There are no overrides applied, so this should be removed.

File: packages/eslint-config/next.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        project,
+      },
+    },
+  },
+  ignorePatterns: [
+    // Ignore dotfiles
+    ".*.js",
+    "node_modules/",
+  ],
+  overrides: [{ files: ["*.js?(x)", "*.ts?(x)"] }],
```
Comment (kylegach) [inline packages/eslint-config/next.js:34]: There are no overrides applied, so this should be removed.

File: packages/eslint-config/react-internal.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    },
+  },
+  ignorePatterns: [
+    // Ignore dotfiles
+    ".*.js",
+    "node_modules/",
+    "dist/",
+  ],
+  overrides: [
+    // Force ESLint to detect .tsx files
```
Comment (kylegach) [inline packages/eslint-config/react-internal.js:36]: Uhh....

File: packages/eslint-config/next.js (hunk 11 above)
Comment (kylegach) [reply packages/eslint-config/next.js:34]: But first, look into https://github.com/storybookjs/web/pull/34/files#r1592835177

File: packages/utils/src/docs-versions.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+/**
+ * ********************************************
+ * ** List of versions of the documentation  **
+ * ********************************************
+ **/
+
+/** Latest version must ALWAYS be first */
+export const docsVersions: DocsVersion[] = [
```
Comment (kylegach) [inline packages/utils/src/docs-versions.tsx:18]: This is now the third place this has been defined.

File: apps/frontpage/lib/get-page.tsx [hunk 14]
```diff
@@ -72,9 +72,13 @@ export const getPageData = async (
         <MDX.CodeSnippets activeVersion={activeVersion.id} {...props} />
       ),
       Callout: MDX.Callout,
+      If: MDX.If,
       IfRenderer: MDX.IfRenderer,
       YouTubeCallout: MDX.YouTubeCallout,
       FeatureSnippets: MDX.FeatureSnippets,
+      HomeRenderers: MDX.HomeRenderers,
+      HomeConcepts: MDX.HomeConcepts,
+      HomeResources: MDX.HomeResources,
```
[the remark this answers is not in the document]
Comment (cdedreuille) [reply apps/frontpage/lib/get-page.tsx:82]: That's true. We should refactor it in a separate PR.

File: apps/frontpage/postcss.config.mjs (hunk 6 above)
Comment (cdedreuille) [reply apps/frontpage/postcss.config.mjs:1]: Yes apparently this is the new way of doing it.

**Pull request merged**